### PR TITLE
Delete and edit

### PR DESCRIPTION
**Pull request by Froggerson**
Folder:
image
Card:
image

### CODE REVIEW

File: frontend/src/flashcards/Flashcard.js [hunk 1]
```diff
@@ -30,74 +26,67 @@ const imageStyle = {
...
+				<ReactCardFlip
+					isFlipped={this.state.isFlipped}
+					flipDirection="horizontal"
+					containerStyle={flipStyle}
+					cardZIndex="-2">
+					<FlashcardFront text={this.props.text} image={this.props.image} />
+					<FlashcardBack text={this.props.translation} />
+				</ReactCardFlip>
+				<div className="icons" style={iconStyle}>
+					<Link
```
Comment (tylerhou) [inline frontend/src/flashcards/Flashcard.js]: Run prettier with `--tab-width 2`

**Review by tylerhou — COMMENTED** [review]
Hard to tell what's going on here because of the formatting changes.

Also, tests?

File: frontend/src/flashcards/Flashcard.js (hunk 1 above)
Comment (Froggerson) [reply frontend/src/flashcards/Flashcard.js]: Alright!

**Comment by Froggerson** [comment]
> Hard to tell what's going on here because of the formatting changes.
> 
> Also, tests?

I think tests for the delete and edit servlets were approved in another PR made by Thomas.

**Comment by tylerhou** [comment]
I think you can write frontend tests?

File: frontend/src/flashcards/Flashcard.js [hunk 2]
```diff
@@ -45,26 +41,19 @@ class Flashcard extends React.Component {
...
-    await fetch("deletecard", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({ cardKey: this.props.cardKey }),
-    });
+    const params = new URLSearchParams();
+    params.append("cardKey", this.props.cardKey);
+    await fetch("/deletecard", { method: "POST", body: params });
```
Comment (tylerhou) [inline frontend/src/flashcards/Flashcard.js:47]: `params.toString()`

File: frontend/src/main-components/MyFoldersContent.js [hunk 3]
```diff
@@ -30,15 +29,14 @@ class MyFoldersContent extends React.Component {
...
-      const foldersData = await fetch("/userfolders")
-        .then((result) => result.json())
-        .then((data) => data["userFolders"]);
-      this.setState({ isDataFetched: true, folders:foldersData });
-    } catch (err) {
-      logDebugMessage("Folders can not be fetched", debugMessageOn);
-    }
+    const foldersData = await fetch("/userfolders")
+      .then((result) => result.json())
+      .then((data) => data["userFolders"]);
```
Comment (tylerhou) [inline frontend/src/main-components/MyFoldersContent.js:34]: Don't mix async and Promise syntax — use one or the other.

**Review by tylerhou — APPROVED** [review]

**Pull request merged**